### ISSUE TITLE
No playback if Fluendo's gst codec pack is installed

### ISSUE THREAD

**Issue by ahjolinna** [alamminsalo/orion#34]
so I just installed the [Fluendo's gstreamer codec pack](http://www.fluendo.com/en/products/multimedia/oneplay-codec-pack/) and after that orion wouldn't work anymore, other gst based players seem to work just fine.

you have to keep in mind that I use openSUSE leap 42.2. and it uses Qt5.6.1 instead of Qt5.7.

here is the gdb output: [link](http://pastebin.com/raw/z9WE0eMQ)
only thing new was: `warning: Cannot parse .gnu_debugdata section; LZMA support was disabled at compile time`

I think it seems to be "running" but it just wont show the playback.

---------------------

the codec pack is 20€/year, if you dont want/cant spend the money I could send a copy (there is prebuild rpm and deb version)

**Comment by zabuldon** [comment]
Hello,
Can you provide with some information?
1. Which backedn are you using for orion? (QtAV, MPV, QtMultimedia)
2. Can I get copy of rpm?

I'm already installed openSUSE to check this issue.

**Comment by ahjolinna** [comment]
1.) I use QtMultimedia ...as live video streaming with MPV doesnt work that well on linux (mostly ffmpeg issue), and QtAV...well I dont want to install yet another player just for orion...if the backend was splitted then I could use it....also I dont know if its also affected by the same ffmpeg issue.

2.) is this your email? vortexius@gmail.com  if so I can send the rpm pkgs there

**Comment by zabuldon** [comment]
Yes, its my emeail. Regarding the download link, I can't download it (asking for registration).

**Comment by ahjolinna** [comment]
@zabuldon : it's sended ...I hope you got the email, check spam folder if you don't see it.

hopefully you will find the problem

**Comment by ahjolinna** [comment]
@zabuldon : I have now left openSUSE, after they updated the nvidia driver to broken one (even nvidia said that nobody should use it)...so I couldn't boot to my opensuse installation ....thanks to this... oh openSUSE respons in IRC was in short "sucks to be you" ...."when you install the NVIDIA proprietary drivers you just modified your openSUSE install into a non-supported state" ...so yeah in short it was like F#ck you....not our fault that we don't check this stuff beforehand :D

ANYWAY... I'm now using manjaro-kde and test this fluendo codec again, and it works...but the loading takes REALLY longtime, and when there isn'ẗ any real UI indicator for something is happening in background its really frustrating...but at least the video/audio quality is better now

**Comment by zabuldon** [comment]
@ahjolinna so, i'm still checking. I'll find cause of this anyway, because its looks really strange.

**Comment by ahjolinna** [comment]
@zabuldon : any luck? it seems that it doesnt work now  at all on openSUSE, at least what I notice when I'm currently working to make my [own openSUSE spinoff/fork](https://susestudio.com/a/kVQm7Y/ahjolinna-plasma-next) so it would be nice to get this working...

I doubt mpv version will work as mpv and (default) ffmpeg is really lightweight version, I don't know why they like "minimal" version with no features...(I will try to build my own mpv version soon)

here is the lastest [strace log](http://pastebin.com/raw/ZPKKVQ78)
[and the pkg I maintain](https://build.opensuse.org/package/show/home:ahjolinna/orion)

@alamminsalo : do u have any ideas....

-----

PS. only thing that works, is the audio-only "quality option in VOD's

**Comment by ahjolinna** [comment]
@alamminsalo & @zabuldon  : here is the[ fluendo's bug report/ticket](http://support.fluendo.com/support/tickets/2108) it has new logs with GST_DEBUG environment variable

**Comment by ahjolinna** [comment]
@zabuldon @alamminsalo : I finally got a respond from fluendo: 

> Hi Alexander,
> 
> Ticket: http://support.fluendo.com/helpdesk/tickets/2108
> 
> We managed to reproduce the issue on a physical machine. Unfortunately, it seems to be related to the way the stream is passed to the decoder. It contains no width, height or framerate information in the caps, and it does not contain SPS/PPS inside the stream, so it is unusable.
> If you could pass this information to the developers of Orion, they might do something about it.
> 
> Regards,
> Xavi

**Comment by alamminsalo** [comment]
Alright, thanks for going through the trouble. I believe this issue can finally be fixed with this information

**Comment by ahjolinna** [comment]
I don't know personally, so I just asked Fluendo if they know/can help. ...Have you tried asking Qt community ([irc channel](https://wiki.qt.io/Online_Communities)) if they would know?

**Comment by alamminsalo** [comment]
No I haven't but that could be a good place to start. On the other hand it's absolutely possible that twitch doesn't serve these headers since they primarily support their own players, but I can't know for sure since admittedly I'm not a video format expert

**Comment by ahjolinna** [comment]
@alamminsalo : I dont know if u have check out gstreamer documentation at all, if u havent...
here is about [debug](https://gstreamer.freedesktop.org/documentation/tutorials/basic/debugging-tools.html) and [streaming](https://gstreamer.freedesktop.org/documentation/tutorials/basic/streaming.html) and [Progressive streaming](https://gstreamer.freedesktop.org/documentation/tutorials/playback/progressive-streaming.html) and [Hardware-accelerated video decoding](https://gstreamer.freedesktop.org/documentation/tutorials/playback/hardware-accelerated-video-decoding.html)


my logs have only used "GST_DEBUG="*:2" orion" ...(the 2 can be replaced with 1-8 like the documentation said)
